src/rebar3_hex_docs.erl

```diff
@@ -156,6 +157,32 @@ resolve_doc_dir(AppInfo) ->
     Dir = proplists:get_value(dir, EdocOpts, ?DEFAULT_DOC_DIR),
     proplists:get_value(doc, AppDetails, Dir).
 
+maybe_gen_docs(State) -> 
+    case doc_opts(State) of
+        undefined -> 
+             rebar_api:error("No docs config found", []),
+            noop;
+        PrvName -> 
+            AllProviders = rebar_state:providers(State),
+             case providers:get_provider(PrvName, AllProviders) of
```

Comment:
We may want to abstract the user of providers in rebar, maybe in rebar_api.

Reply:
I don't think that this would be necessary. If we change from providers we break all plugins regardless since they all use it directly to instantiate themselves.